packages/devextreme-react/src/core/helpers.ts

```diff
@@ -2,10 +2,10 @@
 import { createContext } from 'react';
 import { TemplateInstantiationModel, UpdateLocker } from './types';
 
-export const RemovalLockerContext = createContext<UpdateLocker>({
-  lock: () => undefined,
-  unlock: () => undefined,
-});
+export const RemovalLockerContext = createContext<UpdateLocker | undefined>(undefined);
+
+// eslint-disable-next-line @typescript-eslint/no-extra-parens
+export const RestoreTreeContext = createContext<(() => void) | undefined>(undefined);
```

Comment:
won't just <undefined | () => void > work
but i see that undefined usually defined at the end, just a comment

Reply:
I tried both of course, and both `undefined | () => void` and `(() => void) | undefined` trigger ESLint errors. I just find the first one better looking.

We also don't permit using the `Function` alias.

Of course, I could use a type name for `() => void` and say `<TypeName | undefined>`, but this looks clumsy.

To be honest, part of the reason I left the `// eslint-disable` in this PR is to bring your attention to this issue and discuss maybe we need to ease ESLint rules a bit.